### ISSUE TITLE
The first call to requestPermission() will directly return false

### ISSUE THREAD

**Issue by shinexoh** [santhosh-D-subramani/Shizuku-Plugin#2]
If the user has not granted permission to `Shizuku`, calling the `requestPermission` function will directly return false; it does not wait for the user to choose whether to allow `Shizuku` permission in the pop-up window before returning the corresponding `true` or `false`, but instead returns `false` in advance at the time of the call.

Reproduce code
```dart
if (await shizukuApi.pingBinder() ?? false) {
final result = await shizukuApi.requestPermission() ?? false;
print(result); // false
}
```

@santhosh-D-subramani

**Comment by shinexoh** [comment]
I had AI fix the Java code in the packages, and it successfully corrected the error. The image below shows the modified code by AI, hoping it can inspire you a bit. Since I'm not very familiar with Java, I didn't directly submit a PR :)

![image](https://github.com/user-attachments/assets/227e7adb-4b54-4bc4-99d2-137abb103336)

**Comment by santhosh-D-subramani** [comment]
fixed this in 1.2.1, check on it.

**Comment by shinexoh** [comment]
Everything is normal, thank you
